packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -114,6 +114,35 @@ const networkConnectionWarningToast = () =>
     { autoClose: false }
   )
 
+// Returns false if user rejected the network switch and network is still incorrect
+async function ensureCorrectNetwork({
+  switchChainAsync,
+  sourceChainId,
+  currentChainId
+}: {
+  switchChainAsync: ReturnType<
+    typeof useSwitchNetworkWithConfig
+  >['switchChainAsync']
+  sourceChainId: number
+  currentChainId: number | undefined
+}) {
+  let newChain: { id: number } | null = { id: currentChainId || -1 }
+  while (newChain?.id !== sourceChainId) {
+    try {
+      newChain = await switchChainAsync({
+        chainId: sourceChainId
+      })
+    } catch (e) {
```

Comment:
What if it failed for some other reason? I think we should return `false` always in the `catch`